language-testutils/src/test/java/de/jplag/testutils/LanguageModuleTest.java

```diff
@@ -0,0 +1,273 @@
+        }
+    }
+
+    /**
+     * Returns all test sources, that need to be checked for a matching token sequence
+     * @return The test sources
+     */
+    final List<TestDataCollector.TokenListTest> testTokenSequenceData() {
+        return addNullGuard(this.collector.getTokenSequenceTest());
+    }
+
+    /**
+     * Tests all configured test sources for a monotone order of tokens
+     * @param data The test source
+     * @throws ParsingException If the parser throws some error
+     * @throws IOException If any IO Exception occurs
+     */
+    @ParameterizedTest
+    @MethodSource("getAllTestData")
+    final void testMonotoneTokenOrder(TestData data) throws ParsingException, IOException {
+        if (data != null) {
+            List<Token> extracted = data.parseTokens(this.language);
+
+            for (int i = 0; i < extracted.size() - 2; i++) {
+                Token first = extracted.get(i);
+                Token second = extracted.get(i + 1);
+
+                if (first.getLine() > second.getLine()) {
+                    fail(String.format("Invalid token order. Token %s has a higher line number (%s) than token %s (%s).", first.getType(),
+                            first.getLine(), second.getType(), second.getLine()));
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks that all configured test sources end with a FileEnd token
+     * @param data The test source
+     * @throws ParsingException If the parser throws some error
+     * @throws IOException If any IO Exception occurs
+     */
+    @ParameterizedTest
+    @MethodSource("getAllTestData")
+    final void testTokenSequencesEndsWithFileEnd(TestData data) throws ParsingException, IOException {
+        if (data != null) {
+            List<Token> extracted = data.parseTokens(this.language);
+
+            Assertions.assertEquals(SharedTokenType.FILE_END, extracted.get(extracted.size() - 1).getType(),
+                    "Last token in " + data.describeTestSource() + " is not file end.");
+        }
+    }
+
+    /**
+     * Returns all configured test sources
+     * @return The test sources
+     */
+    final Set<TestData> getAllTestData() {
+        return addNullGuard(this.collector.getAllTestData());
+    }
+
+    /**
+     * Collects the test sources
+     */
+    @BeforeAll
+    final void collectTestData() {
+        collectTestData(this.collector);
+    }
+
+    /**
+     * Adds null to the data, if it is empty
+     * @param data The list to modify
+     * @param <T> The type of items
+     * @param <C> The collection type
+     * @return The modified list
+     */
+    private <T, C extends Collection<T>> C addNullGuard(C data) {
+        if (data.size() == 0) {
+            data.add(null);
+
+            logger.info("No tests configured. Creating a default null test, to fulfill JUnits requirements.");
+        }
+
+        return data;
+    }
```

Comment:
From the comments, I cannot understand why we use this guard. Why do we need it?

Reply:
In which cases do we even need 0 executions? Don't we always have at least one?

Reply:
I see. But this then means there is always one test shown as executed per test type, even though there actually is none, right?

Reply:
e.g. if we add the line
```java
testContainers.add(DynamicContainer.dynamicContainer("TEST", List.of()));
```
after line 88 in `de.jplag.endtoend.EndToEndSuiteTest` we get the following:

![image](https://github.com/jplag/JPlag/assets/4396919/720d8783-5f2f-4a90-95f4-2a44fd970e0e)